### PR TITLE
Feature: Add ability to deactivate user

### LINKED ISSUE #1777: Deactivate editors
As admin
I would like to deactivate editors.
in order to use my paid seats properly.

### PR DESCRIPTION
**Pull request by mattwr18**
- When deactivated user attempts to sign in, redirect with alert message

Fixes #1777

### CODE REVIEW

File: config/locales/de.yml [hunk 1]
```diff
@@ -680,6 +680,7 @@ de:
   sessions:
     errors:
       otp_incorrect: "Der 6-stellige Anmeldecode ist nicht korrekt."
+      deactivated_user: Dein Konto ist derzeit nicht aktiv, bitte kontaktierst du den Support unter support@tactile.news.
```
Comment (mattwr18) [inline config/locales/de.yml:683]: @drjakob can you ok this is what you want to say?

Comment (drjakob) [reply config/locales/de.yml:683]: "Dein Konto ist derzeit nicht aktiv, bitte kontaktiere den Support unter support@tactile.news"

File: spec/support/shared_examples/activity_notifications.rb [hunk 2]
```diff
@@ -10,8 +10,8 @@
 
     it 'for each user' do
       run_action(subject)
```
Comment (soey) [inline spec/support/shared_examples/activity_notifications.rb:12]: will an inactive user get notifications and what is expected in that regard?

**Comment by soey** [comment]
Well this is one of those features that goes through the whole application ... you have to review all collection retrievals of users and have to consider which scope to be used. I've yet no overview on if there are more occurences somewhere..

File: app/models/user.rb [hunk 3]
```diff
@@ -27,6 +29,22 @@ def avatar?
     false
   end
 
+  def active?
+    deactivated_at.nil?
+  end
+
+  alias active active?
```
[the remark this answers is not in the document]
Comment (mattwr18) [reply app/models/user.rb:36]: Good question. To be honest, I copied and pasted the implementation for deactivated contributors for the most part, but even for contributors this these aliases don't seem to be used. There is a unit test for `contributor.active`, but it's not used in the application code. I prefer using predicate methods when I expect a boolean returned.

File: spec/support/shared_examples/activity_notifications.rb (hunk 2 above)
Comment (mattwr18) [reply spec/support/shared_examples/activity_notifications.rb:12]: Yes, good point. I'm not sure what the behavior should be in that case. As we discussed in our pairing session, the notifications are only internal to the site and are not sent by any messenger or email so they would not be able to see them unless someone were to reactivate them. I'm not sure exactly what the client would expect to occur in that case. I could see reasons for them to want the user to see, or not see, the notifications.

I'll ask @drjakob about it when I next speak with him.

**Pull request merged**